src/hiero_sdk_python/staking_info.py

```
@@ -0,0 +1,132 @@
+
+    def __post_init__(self) -> None:
+        if self.staked_account_id is not None and self.staked_node_id is not None:
+            raise ValueError("Only one of staked_account_id or staked_node_id can be set.")
+
+    @classmethod
+    def from_proto(cls, proto: StakingInfoProto) -> "StakingInfo":
+        """
+        Creates a StakingInfo instance from its protobuf representation.
+        """
+        if proto is None:
+            raise ValueError("Staking info proto is None")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)